src/azul/service/repository_service.py

```diff
@@ -70,12 +69,28 @@ def __init__(self, entity_type: str, entity_id: str):
         super().__init__(f"Can't find an entity in {entity_type} with an uuid, {entity_id}.")
 
 
+@attrs.frozen(auto_attribs=True, kw_only=True)
 class SearchResponseStage(_ElasticsearchStage[ResponseTriple, MutableJSON],
                           metaclass=ABCMeta):
+    file_url_func: FileUrlFunc
 
     def prepare_request(self, request: Search) -> Search:
         return request
 
+    @property
+    def repository_plugin(self) -> RepositoryPlugin:
+        return self.service.repository_plugin(self.catalog)
+
+    def _file_url(self, uuid: str, version: str, drs_uri: str | None) -> str | None:
```

Comment:
Because they are all strings they could accidentally be swapped:

```suggestion
    def _file_url(self, *, uuid: str, version: str, drs_uri: str | None) -> str | None:
```